scripts/bottle-hashes.sh

```diff
@@ -19,10 +19,13 @@ then
                 os="${BASH_REMATCH[2]}"
                 formula_file="./Formula/$formula_name.rb"
                 if [[ -f $formula_file ]]; then
+                    formula_tag="$(sed -n "s/^\s\+version \"\(.*\)\"/\1/p" "$formula_file")"
                     line="\    sha256 cellar: :any, $os: \"$bottle_hash\""
                     # Update only when this formula doesn't have a hash for the bottle
-                    if ! grep "$line" "$formula_file" &> /dev/null; then
+                    if [[ "$formula_tag" == "$2" ]]; then
```

Comment:
Related to the point above: instead of checking this here, aka when trying to update the hashes, couldn't we check this earlier and avoid the bottle where this doesn't match from being built in the first place?

Not sure if it would be any better, in fact AFAIK the system that already checks for existing bottles should work quite well in preventing rebuilds, but I wanted to mention/ask about it

Reply:
The problem is that GA spawns a job per bottle and there is already a check that prevents building the bottle in case it was already built. And IMO such an approach works a bit better when there are some flaky build errors